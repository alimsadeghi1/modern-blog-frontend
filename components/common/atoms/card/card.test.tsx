import React from "react";
import { render } from "@testing-library/react";
import CardComponent from "./card.component";
import AppProviderComponent from "@/components/providers/app/app-provider.component";
import "@testing-library/jest-dom";

describe("CardComponent", () => {
  it("renders with default props", () => {
    const { container } = render(
      <AppProviderComponent>
        <CardComponent />
      </AppProviderComponent>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with custom props", () => {
    const { container } = render(
      <AppProviderComponent>
        <CardComponent data-testid="card" className="custom-class" />
      </AppProviderComponent>
    );
    expect(container.firstChild).toHaveClass("custom-class");
    expect(container.firstChild).toHaveAttribute("data-testid", "card");
  });
});
