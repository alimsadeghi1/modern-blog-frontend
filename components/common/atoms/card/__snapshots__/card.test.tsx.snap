// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent renders with default props 1`] = `
<div
  class="css-1iiy8m6"
/>
`;
